src/helpers/path.js

```diff
@@ -9,16 +12,23 @@
  * @param {number} targetY - target y location (in cells)
  * @param {number} grid - 2D array of cells. Traversable cells are 1s, others are 0.
  */
-export function getShortestPath(myX, myY, targetX, targetY, grid) {
+export function getShortestPath(myX, myY, targetX, targetY, grid, diagonal = false) {
   // TODO: handle edge cases regarding target and current position
   // diagonal is true if we consider diagonal steps on the grid
-  const diagonal = false;
+
+  assertGridInBounds(grid, myX, myY);
+  assertGridInBounds(grid, targetX, targetY);
+
   const graph = new Graph(grid, { diagonal });
   const start = graph.grid[myX][myY];
   const end = graph.grid[targetX][targetY];
   // calculate shortest path list
   const shortestPath = astar.search(graph, start, end,
     { heuristic: diagonal ? astar.heuristics.diagonal : astar.heuristics.manhattan });
+
+  if (shortestPath.length === 0) {
+    return undefined;
```

Comment:
Is this just to make the return more explicit? This feels like a weird way to handle this, though given the removal of the TODO I imagine that it fixes the issue arising during the 0.25s when the flag doesn't exist. Could you elaborate why this fixes that?